### PR TITLE
[#156371588] Add redux setup

### PR DESCRIPTION
**Pull request by Descartess**
### What does this PR do?
Implements  redux set up.

### Description of Task to be completed?
 - Added redux boilerplate
 - Added redux persist of store to async storage

### How should this be manually tested?
- Run the application using `react-native run-android` or `react-native run-ios`  to see the changes.
- Enable debugging Js remotely on your terminal 
- Navigate within the app and watch the state changes logged on console.

### Any background context you want to provide?
This is basic project set up, hoping to be basis to start building more features.

#### What are the relevant pivotal tracker stories?
[#156371588](https://www.pivotaltracker.com/story/show/156371588)
### Screenshot
redux-persist

### CODE REVIEW

File: src/reducers/index.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+import { combineReducers } from 'redux'
+import counterReducer from './counterReducers'
+
+const reducers = combineReducers({
+    counter: counterReducer
+})
+
+export default reducers
```
Comment (sirjmkitavi) [inline src/reducers/index.js]: @Descartess you may be add a new line at end of the files.

File: package.json [hunk 2]
```diff
@@ -9,15 +9,25 @@
...
+    "react-navigation": "^1.5.8",
+    "react-redux": "^5.0.7",
+    "redux": "^3.7.2",
+    "redux-persist": "^5.9.1",
+    "redux-promise-middleware": "^5.0.0"
   },
   "devDependencies": {
     "babel-jest": "22.4.3",
     "babel-preset-react-native": "4.0.0",
+    "eslint": "^4.19.1",
```
Comment (sirjmkitavi) [inline package.json]: could the eslint configurations come in as a different PR??

File: src/types/counterTypes.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export const INCREMENT = "INCREMENT"
```
Comment (sirjmkitavi) [inline src/types/counterTypes.js]: Why have you placed the action types outside of the actions folders?
I thought it's standard to have them inside the actions folder... just my thoughts.

Comment (sirjmkitavi) [reply src/types/counterTypes.js]: @Descartess @ivantoz This is still pending.
Am not sure what you guys think???

Comment (ivantoz) [reply src/types/counterTypes.js]: On it @sirjmkitavi

File: src/components/Login/index.js [hunk 4]
```diff
@@ -24,4 +24,5 @@ class Login extends Component {
   }
 }
 
+
```
[the remark this answers is not in the document]
Comment (Descartess) [reply src/components/Login/index.js]: Why are we using redux integration with react navigation, Is there a particular  use case that requires it. i.e Some navigation that we can’t achieve without redux integration.The react navigation docs advise against redux integration  .See here https://reactnavigation.org/docs/redux-integration.html. Check warning section. @ivantoz @sirjmkitavi

**Comment by Descartess** [comment]
Why are we using redux integration with react navigation, Is there a particular use case that requires it. i.e Some navigation that we can’t achieve without redux integration.The react navigation docs advise against redux integration .See here https://reactnavigation.org/docs/redux-integration.html. Check warning section. @ivantoz @sirjmkitavi

**Pull request merged**